### ISSUE TITLE
NSFW Detection

### ISSUE THREAD

**Issue by harshita-srivastava-yral** [yral-dapp/off-chain-agent#74]

**Comment by harshita-srivastava-yral** [comment]
- Fly token is required from Saikat to deploy it on production

**Comment by harshita-srivastava-yral** [comment]
- @jay-dhanwant-yral to collaborate with @komal-sai-yral for the next steps on this

**Comment by harshita-srivastava-yral** [comment]
- Connected with @komal-sai-yral 
- Need to qrite few queries but that has dependency on Komal's end for a task to be done

**Comment by harshita-srivastava-yral** [comment]
- Integration task to be done with @komal-sai-yral today
- Post that ETA to be shared on NSFW task

**Comment by harshita-srivastava-yral** [comment]
- Waiting for Komal to get back with the pipeline

**Comment by harshita-srivastava-yral** [comment]
- Made the changes for NSFW and working on local
- Prod key wasn't working as FLY token was failing

**Comment by harshita-srivastava-yral** [comment]
- Pushed things to main fly account.

**Comment by harshita-srivastava-yral** [comment]
- We would want to have cleaner feed on Yral domain
- We would want to work on NSFW first on Yral before the comments section

**Comment by harshita-srivastava-yral** [comment]
- Completed with NSFW implementation on the pumpAI
- Need a fly token for the production deployment
- Post this, connect with @komal-sai-yral 
- Pick up Yral feed as next project

**Comment by harshita-srivastava-yral** [comment]
- We will have NSFW feed only in hotornot.wtf domain
- We need a clean version of the app where only cleaner videos are shown and doesn't have any toggle
- This app will be used for app store submission and interaction with financial institution

**Comment by harshita-srivastava-yral** [comment]
- Further enhancement of meta data is being done on the current feed
- Provocative tagged and popularity as metrics were considered
- Selectively give more importance to certain dimensions

**Comment by harshita-srivastava-yral** [comment]
- 2 Exercises done - Softmax function used to test the video embedding gave 2-3% higher lift. Post training the model we were able to get 95% accuracy. Recommendation is to go ahead with this
- 70% accuracy for the grey area provocative NSFW videos 
- We should be 90% accuracy overall
- Disclaimer - Training data is less currently. We don't have rich annotation
- We might have to train again at a large data. 
- [ ] We should do retraining on entire data.

**Comment by harshita-srivastava-yral** [comment]
- We are half way there

**Comment by harshita-srivastava-yral** [comment]
- We have the annotations done
- We have testing done on new model and its trained
- We got some validation and we have report to share

**Comment by harshita-srivastava-yral** [comment]
- Prod readiness of NSFW model
- Once Komal is back, we will backfill all the data

**Comment by harshita-srivastava-yral** [comment]
- NSFW videos have been popping up which needs investigation from @jay-dhanwant-yral

**Comment by jay-dhanwant-yral** [comment]
- I checked the nsfw videos that were popping up in the platform. They were false negatives from the previous vision based  models. 
- I ran the same videos with the embedding based model which is going to be live soon, and I didn't get any false negatives.

**Comment by jay-dhanwant-yral** [comment]
TODOs for embedding deployment 
-[] Make the data pipeline live through dataproc 
-[] Finalize model hyper-parameters (squeeze out the last bit of accuracy)
-[] Decide and schedule on re-training
-[] Alerts for model degradation
-[] Take the model live if the `benchmarking_df` has a good enough accuracy

**Comment by harshita-srivastava-yral** [comment]
- Embedding based model we are on track

**Comment by harshita-srivastava-yral** [comment]
- Deployment to be done today

**Comment by harshita-srivastava-yral** [comment]
- DS side we are done we need to connect with @komal-sai-yral on how to integrate that into system

**Comment by siyara-m-yral** [comment]
- Video embedding should be done via offchain to avoid expensive Joins. @komal-sai-yral to pick related task
